Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,211 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using Modix.Common.Messaging;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Promotions;
+using Modix.Data.Repositories;
+using Modix.Services.Core;
+using Modix.Services.Images;
+using Modix.Services.Promotions;
+using Modix.Services.Utilities;
+
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogBehavior :
+        INotificationHandler<ReactionAddedNotification>,
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly TimeSpan ErrorRemoveDelay = TimeSpan.FromSeconds(5);
+
+        private readonly DateTime _utcNow = DateTime.UtcNow;
+
+        private static readonly Emoji _approveEmoji = new Emoji("👍");
+
+        private static readonly Emoji _disapproveEmoji = new Emoji("👎");
+
+        private IDesignatedChannelService _designatedChannelService { get; }
```

Comment:
is there a reason you prefer a field over a prop?

Reply:
Speaking for myself, private auto-properties feel strange and kind of pointless. I can't think of any examples of well-known codebases that use private properties either.

Reply:
Because of the IL it emits.

```cs
private object MyProperty { get; }
```

is literally equivalent to 

```cs
private object MyProperty
{
    get
    {
        return _myProperty;
    }
}
private readonly object _myProperty;
```

All you're doing is wasting CPU cycles.

Additionally, the main reasons that properties are preferred over fields for `public` and `protected` members are as follows:

A) Properties are more easily reflected upon. In fact, many serializers are built to recognize properties, rather than fields, and the WPF data-binding system requires properties specifically.

B) Using a property for something that doesn't actually have any get or set logic may seem redundant, but it allows for you to add logic later, without causing a breaking change of the class's surface area. That is, If something is implemented as a field, and you later change it to be a property instead, so you can add some logic within a custom getter or setter, you've changed the surface area of your class, and (related to A) this might break consumers.

Since `private` members have no consumers, both of these points are moot.